boa3/internal/analyser/astanalyser.py

```diff
@@ -47,6 +49,9 @@ def has_errors(self) -> bool:
         return len(self.errors) > 0
 
     def _log_error(self, error: CompilerError):
+        if self._fail_fast and len(self.errors) > 0:
```

Comment:
Just to make sure I correctly understood, this first `if` will only be triggered if another Analyser already logged the correct error, while the last `if` will only be triggered if this analyser logs the error?

Reply:
this first if is here because when the generic `visit` has a try except that handles CompilerError, so it won't directly raise the logged error (it'll be catch from the outer `visit` call)
so, in generic calls for `visit` the first log_error call will raise de exception in the second if and all the following will end in this first if